src/Sentry/SentryOptions.cs

```diff
@@ -1297,6 +1297,10 @@ public SentryOptions()
                 UriComponents.SchemeAndServer,
                 UriFormat.Unescaped)
         );
+
+#if PLATFORM_NEUTRAL
```

Comment:
What does this mean in practice? Did we document in what environments this will work and which ones it wont?

Reply:
It's the flip side to this:
https://github.com/getsentry/sentry-dotnet/blob/b76d536f15cfad9cf19ad0e218e1914b58a9e403/src/Sentry.Maui/Internal/SentryMauiOptionsSetup.cs#L54-L57

With `PLATFORM_NEUTRAL` being [defined here](https://github.com/getsentry/sentry-dotnet/blob/199cda3e3f06baeae3f5e7b286c1b7edc0996dd2/Directory.Build.props#L55-L67).

In practice, I think that means if you lose network connectivity and then regain it again, and you're targeting one of those platforms specifically but you're not using MAUI, then any envelopes that you tried to send while the network was down won't be sent until you restart the application again. 

Another option would be to **_always_** initialise the `SentryOptions.NetworkStatusListener` to a `new PollingNetworkStatusListener` but let the `SentryMauiOptionsSetup` override this... or some variant of that. Technically `SentryOptions.NetworkStatusListener` is public so the user could set this to some custom listener that they've crafted. We don't currently do anything to make sure we don't override the user's listener if they've already set one.

Reply:
Have created https://github.com/getsentry/sentry-dotnet/issues/3510 to track this conversation in any case (this PR is closed/merged so easy for the conversation to get lost here).

Reply:
> We don't currently do anything to make sure we don't override the user's listener if they've already set one.

I think that's the crux. But if we set it up in the option's constructor the user would overwrite it again and we'd be fine, right?